homeassistant/components/rflink/sensor.py

```diff
@@ -164,7 +233,10 @@ async def async_added_to_hass(self) -> None:
     @property
     def native_unit_of_measurement(self):
         """Return measurement unit."""
-        return self._unit_of_measurement
+        if self._unit_of_measurement:
+            return self._unit_of_measurement
+        if hasattr(self, "entity_description"):
+            return self.entity_description.native_unit_of_measurement
```

Comment:
Looks to be missing an explict return None in the failing case.

Reply:
Done.